src/printer.js

```diff
@@ -3678,11 +3678,17 @@ function isBinaryish(node) {
 }
 
 function shouldInlineLogicalExpression(node) {
-  return (
-    node.type === "LogicalExpression" &&
-    (node.right.type === "ObjectExpression" ||
-      node.right.type === "ArrayExpression")
-  );
+  if (node.type === "LogicalExpression" &&
```

Comment:
Can you early return false if that's a logical expression, this way you don't need to have it written twice.